### PR TITLE
Update types for swagger-jsdoc 6.0

### PR DESCRIPTION
**Pull request by preyansh07**
* Update swagger-jsdoc types to version 6.0
* Add Oas3Options and Oas3Definition interfaces for [swagger-jsdoc options](https://github.com/Surnet/swagger-jsdoc/blob/7aaa3e00b75c6deffcce09126beee391716549d2/examples/app/app.js#L38) and [OAS 3.0 object](https://swagger.io/specification/#openapi-object) respectively
* Add types for OAS 3.0 schema objects

---

Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] [Run `npm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: [swagger-spec](https://github.com/Surnet/swagger-jsdoc/blob/master/examples/app/swagger-spec.json), [swagger-jsdoc getting started](https://github.com/Surnet/swagger-jsdoc/blob/master/docs/GETTING-STARTED.md#getting-started), [Open API Specification v3](https://swagger.io/specification/#openapi-object)
- [x] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the header.

### CODE REVIEW

File: types/swagger-jsdoc/index.d.ts [hunk 1]
```diff
@@ -44,32 +44,287 @@
 declare function swaggerJSDoc(options?: swaggerJSDoc.Options): object;
 
 declare namespace swaggerJSDoc {
-    interface ApiInformation {
-        description?: string;
+    /**
+     * Open API Specification (OAS) version 3.0 options
+     */
+    interface Oas3Options {
```
Comment (drGrove) [inline types/swagger-jsdoc/index.d.ts:50]: `OAS3Options` as acronyms should always be capitalized

File: types/swagger-jsdoc/index.d.ts [hunk 2]
```diff
@@ -44,32 +44,287 @@
...
+        apis?: ReadonlyArray<string>;
+        definition?: Oas3Definition;
+        swaggerDefinition?: Oas3Definition;
+        [key: string]: any;
+    }
+
+    /**
+     * For describing Open API Specification (OAS) version 3.0
+     */
+    interface Oas3Definition {
```
Comment (drGrove) [inline types/swagger-jsdoc/index.d.ts:60]: OAS3Definition

**Review by drGrove — CHANGES_REQUESTED** [review]

File: types/swagger-jsdoc/index.d.ts (hunk 1 above)
Comment (preyansh07) [reply types/swagger-jsdoc/index.d.ts]: I have made the suggested changes. please review.

**Review by drGrove — APPROVED** [review]

**Comment by preyansh07** [comment]
Ready to merge

**Pull request merged**